circleci.sh

```diff
@@ -97,9 +98,17 @@ coverage()
     make -f posix.mak $(find std etc -name "*.d" | sed "s/[.]d$/.test/" | grep -vE '(std.algorithm.sorting|std.encoding|net.curl)' )
 }
 
+# compile all public unittests separately
+publictests()
+{
+    clone https://github.com/dlang/tools.git ../tools master
```

Comment:
> Why master instead of $base_branch? I'd argue that unconditionally using master is almost always wrong, b/c it's not tied to the phobos version.

Thanks for the advice! :)

> Possible ways to avoid future issues:
> checkout a specific version (v2.072.2) or commit of that file in the tools repo

Okay I have done that.

> maintain that tool in phobos if it's merely designed for phobos

Well it's general in the sense that the script has nothing about Phobos hard-coded (except for the script's name), but I think that would have been the best option.